core/trino-main/src/main/java/io/trino/sql/planner/PlanFragmenter.java

```diff
@@ -572,7 +572,7 @@ public GroupedExecutionProperties visitJoin(JoinNode node, Void context)
             GroupedExecutionProperties left = node.getLeft().accept(this, null);
             GroupedExecutionProperties right = node.getRight().accept(this, null);
 
-            if (node.getDistributionType().isEmpty()) {
+            if (!groupedExecutionEnabled || node.getDistributionType().isEmpty()) {
```

Comment:
update comment in next line

Reply:
I moved the `!groupedExecutionEnabled` check to a different block to avoid confusion.